### PR TITLE
Refactor ID Validation Logic into Utility Function

### LINKED ISSUE #890: Utility function for collection_id validation
Just another case of the item ID validation that was missing. We actually now repeat this code 3/4 times in different modals (with the same typos in the error message, fixed here...), so we should wrap this up into a single utility function for IDs (perhaps with the ability to disable some via bool toggles depending on context). Perhaps you could throw this together after @BenjaminCharmes?

_Originally posted by @ml-evs in https://github.com/datalab-org/datalab/pull/888#discussion_r1777685979_

### PR DESCRIPTION
**Pull request by BenjaminCharmes**
Closes #890

- Creation of a new utility function for ID validation inside field_utils.js
- Replacement of duplicate validation logic in modals with calls to the new utility function.

### CODE REVIEW

**Review by ml-evs — COMMENTED** [review]
Just some minor comments, otherwise looks good

File: webapp/src/field_utils.js [hunk 1]
```diff
@@ -60,3 +60,35 @@ export function createComputedSetterForCollectionField(collection_field) {
...
+    return "ID can only contain alphanumeric characters, dashes ('-'), and underscores ('_').";
+  }
+  if (/^[._-]/.test(id) | /[._-]$/.test(id)) {
+    return "ID cannot start or end with punctuation";
+  }
+  if (id.length < 1 || id.length > 40) {
+    return "ID must be between 1 and 40 characters.";
+  }
+  return "";
+}
```
[the remark this answers is not in the document]
Comment (BenjaminCharmes) [reply webapp/src/field_utils.js:94]: Yeah make sense!

File: webapp/src/field_utils.js [hunk 2]
```diff
@@ -60,3 +60,48 @@ export function createComputedSetterForCollectionField(collection_field) {
...
+ *
+ * @param {string} id - The ID to validate.
+ * @param {string[]} [takenIds=[]] - Array of already taken item IDs.
+ * @param {string[]} [existingIds=[]] - Array of IDs already known to exist.
+ *
+ * @returns {string} An empty string if the ID is valid and not in use, otherwise an error message.
+ *                   If the ID is in use, it returns a string error message with an HTML link to edit the ID.
+ *
+ * @throws {TypeError} If id is not a string when it's not null.
+ *
```
Comment (ml-evs) [inline webapp/src/field_utils.js:75]:
```suggestion
```

It probably won't do this? I'll remove this line then merge

**Review by ml-evs — APPROVED** [review]
Thanks @BenjaminCharmes!

**Pull request merged**